src/bp/migrations/v12_18_1-1614560297-change_dialog_sessions_id_column_type.ts

```diff
@@ -0,0 +1,56 @@
+import * as sdk from 'botpress/sdk'
+import { Migration } from 'core/services/migration'
+
+const TABLE_NAME = 'dialog_sessions'
+const TEMP_TABLE_NAME = 'dialog_sessions_temp'
+
+const migration: Migration = {
+  info: {
+    description: 'Change dialog_sessions id column type',
+    target: 'core',
+    type: 'database'
+  },
+  up: async ({
+    bp,
+    configProvider,
+    database,
+    inversify,
+    metadata
+  }: sdk.ModuleMigrationOpts): Promise<sdk.MigrationResult> => {
+    const db = database.knex as sdk.KnexExtended
+    const { client } = db.client.config
+
+    try {
+      if (client === 'sqlite3') {
+        await db.transaction(async trx => {
```

Comment:
You are not checking if the column type needs to be changed. Migration may fail when executed more than once

Reply:
I'm not sure how it can cause the migration to fail if run multiple times, but, I still agree that it's a better practice to add a check before running the migration.